app/controllers/api/v2/anime_controller.rb

```diff
@@ -2,6 +2,21 @@ module Api::V2
   class AnimeController < ApiController
     caches_action :show, expires_in: 1.hour
 
+    def index
+      anime = Anime.preload(:genres)
+
+      if params.key? :filter
+        search = AnimeSearch.new(params[:filter])
+        if search.valid?
+          anime = search.apply_filters(anime)
+        else
+          return render json: search.errors.full_messages
+        end
+      end
+
+      render json: anime.page(params[:page])
```

Comment:
Kaminari requires that an `order` be used on all paginated things.  You should also probably specify a `per`.